packages/kit/src/core/tsconfig.js

```diff
@@ -0,0 +1,101 @@
+import fs from 'fs';
+import path from 'path';
+import { mkdirp } from '../utils/filesystem.js';
+import { SVELTE_KIT } from './constants.js';
+
+/** @param {string} file */
+const exists = (file) => fs.existsSync(file) && file;
+
+/** @param {import('types').ValidatedConfig} config */
+export function generate_tsconfig(config) {
+	const out = path.resolve(SVELTE_KIT, 'tsconfig.json');
+
+	const user_file = exists('tsconfig.json') || exists('jsconfig.json');
+
+	const paths = {};
+
+	paths['$lib'] = [path.relative(SVELTE_KIT, config.kit.files.lib)];
+	paths['$lib/*'] = [path.relative(SVELTE_KIT, config.kit.files.lib) + '/*'];
+
+	if (user_file) {
+		// we have to eval the file, since it's not parseable as JSON (contains comments)
+		const user_tsconfig_json = fs.readFileSync(user_file, 'utf-8');
+		const user_tsconfig = (0, eval)(`(${user_tsconfig_json})`);
+
+		// we need to check that the user's tsconfig extends the framework config
+		const extend = user_tsconfig.extends;
+		const extends_framework_config = extend && path.resolve('.', extend) === out;
+
+		if (extends_framework_config) {
+			const { paths: user_paths } = user_tsconfig.compilerOptions || {};
+
+			if (user_paths) {
+				const lib = user_paths['$lib'] || [];
+				const lib_ = user_paths['$lib/*'] || [];
+
+				const missing_lib_paths =
+					!lib.some(
+						(/** @type {string} */ relative) => path.resolve('.', relative) === config.kit.files.lib
+					) ||
+					!lib_.some(
+						(/** @type {string} */ relative) =>
+							path.resolve('.', relative) === config.kit.files.lib + '/*'
+					);
+
+				if (missing_lib_paths) {
+					console.warn(
+						`\u001B[1m\u001B[93mYour compilerOptions.paths in ${user_file} should include the following:\u001B[39m\u001B[22m`
+					);
+					const relative = path.relative('.', config.kit.files.lib);
+					console.warn(`{\n  "$lib":["${relative}"],\n  "$lib/*":["${relative}/*"]\n}`);
+				}
+			}
+		} else {
+			let relative = path.relative('.', out);
+			if (relative.startsWith(SVELTE_KIT)) relative = './' + relative;
+
+			console.warn(
+				`\u001B[1m\u001B[93mYour ${user_file} should include the following:\u001B[39m\u001B[22m`
+			);
+			console.warn(`"extends": "${relative}"`);
```

Comment:
This is great! I wonder if people will not know exactly where to add it or realize they can remove everything else. Maybe we could tweak the wording slightly to make those points clearer?

```suggestion
			console.warn(
				`\u001B[1m\u001B[93mYour ${user_file} should extend SvelteKit's. New projects simply have:\u001B[39m\u001B[22m`
			);
			console.warn(`{ "extends": "${relative}" }`);
```